spec.html

```diff
@@ -20670,26 +21469,41 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
           1. Set _constructor_ to ParseText(_constructorText_, |MethodDefinition[~Yield, ~Await]|).
           1. Assert: _constructor_ is a Parse Node.
         1. Set the running execution context's LexicalEnvironment to _classScope_.
+        1. Set the running execution context's PrivateEnvironment to _classPrivateEnvironment_.
         1. Let _constructorInfo_ be ! DefineMethod of _constructor_ with arguments _proto_ and _constructorParent_.
         1. Let _F_ be _constructorInfo_.[[Closure]].
         1. Perform SetFunctionName(_F_, _className_).
         1. Perform MakeConstructor(_F_, *false*, _proto_).
         1. If |ClassHeritage_opt| is present, set _F_.[[ConstructorKind]] to ~derived~.
         1. Perform MakeClassConstructor(_F_).
         1. Perform CreateMethodProperty(_proto_, *"constructor"*, _F_).
-        1. If |ClassBody_opt| is not present, let _methods_ be a new empty List.
-        1. Else, let _methods_ be NonConstructorMethodDefinitions of |ClassBody|.
-        1. For each |ClassElement| _m_ of _methods_, do
-          1. If IsStatic of _m_ is *false*, then
-            1. Let _status_ be PropertyDefinitionEvaluation of _m_ with arguments _proto_ and *false*.
+        1. If |ClassBody_opt| is not present, let _elements_ be a new empty List.
+        1. Else, let _elements_ be NonConstructorMethodDefinitions of |ClassBody|.
```

Comment:
Currently class fields do not provide static semantics for `ClassElementKind`, which is necessary to be included in the result of `NonConstructorMethodDefinitions`. Without specifying `ClassElementKind` for the field productions, they won't be included in this result.

Reply:
I believe this is done now. I've also renamed NonConstructorMethodDefinitions to NonConstructorElements.